test/suites/consolidation/suite_test.go

```diff
@@ -65,7 +66,119 @@ var _ = BeforeEach(func() {
 var _ = AfterEach(func() { env.Cleanup() })
 var _ = AfterEach(func() { env.AfterEach() })
 
-var _ = Describe("Consolidation", func() {
+var _ = FDescribe("Consolidation", func() {
+	Context("LastPodEventTime", func() {
+		var nodePool *karpv1.NodePool
+		BeforeEach(func() {
+			nodePool = env.DefaultNodePool(nodeClass)
+			nodePool.Spec.Disruption.ConsolidateAfter = karpv1.NillableDuration{}
+
+		})
+		It("should update lastPodEventTime when pods are scheduled and removed", func() {
+			var numPods int32 = 5
+			dep := test.Deployment(test.DeploymentOptions{
+				Replicas: numPods,
+				PodOptions: test.PodOptions{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"app": "regular-app"},
+					},
+					ResourceRequirements: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+					},
+				},
+			})
+			selector := labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+			nodePool.Spec.Disruption.Budgets = []karpv1.Budget{
+				{
+					Nodes: "0%",
+				},
+			}
+			env.ExpectCreated(nodeClass, nodePool, dep)
+
+			nodeClaims := env.EventuallyExpectCreatedNodeClaimCount("==", 1)
+			env.EventuallyExpectCreatedNodeCount("==", 1)
+			env.EventuallyExpectHealthyPodCount(selector, int(numPods))
+
+			nodeClaim := env.ExpectExists(nodeClaims[0]).(*karpv1.NodeClaim)
+			lastPodEventTime := nodeClaim.Status.LastPodEventTime
+
+			// wait 10 seconds so that we don't run into the de-dupe timeout
+			time.Sleep(10 * time.Second)
+
+			dep.Spec.Replicas = lo.ToPtr[int32](4)
+			By("removing one pod from the node")
+			env.ExpectUpdated(dep)
+
+			Eventually(func(g Gomega) {
+				nodeClaim = env.ExpectExists(nodeClaim).(*karpv1.NodeClaim)
```

Comment:
Why do you need to do this casting? Isn't this function typed so it shouldn't need an interface cast?

Reply:
It doesn't use generics, uses client.Object, which means it just implements an interface iiuc

Reply:
Seems trivial to change it to use generics?

Reply:
Discussed offline: Can't do generics with member functions so we're pushing this out right now